### ISSUE TITLE
TestNG hangs in debug mode (dreaded 57%)

### ISSUE THREAD

**Issue by ghost** [cbeust/testng-eclipse#55]
Ever since I upgraded the testng-eclipse plugin, I hang at 57% when debugging a test (any test) in Eclipse.  The last entry in the error log appears to be:

`
eclipse.buildId=M20120208-0800
java.version=1.7.0_02
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86, WS=win32, NL=en_US
Framework arguments:  -product org.eclipse.epp.package.java.product
Command-line arguments:  -os win32 -ws win32 -arch x86 -product org.eclipse.epp.package.java.product

Error
Mon Mar 12 17:44:07 EDT 2012
Error

java.net.SocketTimeoutException: Accept timed out
    at java.net.TwoStacksPlainSocketImpl.socketAccept(Native Method)
    at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:398)
    at java.net.PlainSocketImpl.accept(PlainSocketImpl.java:183)
    at java.net.ServerSocket.implAccept(ServerSocket.java:522)
    at java.net.ServerSocket.accept(ServerSocket.java:490)
    at org.testng.remote.strprotocol.BaseMessageSender.initReceiver(BaseMessageSender.java:128)
    at org.testng.eclipse.ui.TestRunnerViewPart.startTestRunListening(TestRunnerViewPart.java:402)
    at org.testng.eclipse.TestNGPlugin.connectTestRunner(TestNGPlugin.java:215)
    at org.testng.eclipse.TestNGPlugin$2.run(TestNGPlugin.java:203)
    at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
    at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
    at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4140)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3757)
    at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
    at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
    at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
    at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
    at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:123)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:344)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:179)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:622)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:577)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1410)
`

I am running:

Eclipse Indigo Service Release 2 (build 20120216-1857)
testng-eclipse plugin 6.4.0.20120308_1617

I do have TestNG 6.4 declared as a Maven dependency (test scope) within my project.  I read online that there can be a conflict between the project and plugin's versions of TestNG.  However, both are at 6.4 (verified this in both Maven and Eclipse) for this project.

Any thoughts?  

Thanks so much!
- Saish

**Comment by SidArc** [comment]
I am also facing the same issue, with no solution still, I am just trying to run a simple TestNG Class using TestNG Test and it is causing the Eclipse to Freeze, I have tried almost every solution talked above, can anyone help me with the solution to this issue.

In the above mentioned comments I have read that removing the breakpoints from TestNG class helped, are we here referring to the class file in the testng-6.8.5 jar or in the class file that we have created because I do not have any breakpoints in my class file but still I am facing this issue.

Can someone help me.
